site/content/blog/2021-04-19-nodes-automation-image.md

```diff
@@ -0,0 +1,56 @@
+title: "Automating Noodes Rotation Using Gitlab Pipelines"
+heading: "Automating Noodes Rotation"
+description: "This blog post describes our journey to automate our nodes rotation process when we have a new AMI release, and the open source tools we built on this."
+slug: cloud-monitoring
+date: 2019-12-18T00:00:00-04:00
+author: Stavros Foteinopoulos
+github: stafot
+community: stavros.foteinopoulos
+---
+#### Overview
+
+In the daily life of a Site Reliability Engineer the main goal is to reduce all the work we call toil. But what is toil? Toil is the kind of work tied to running a production service that tends to be manual, repetitive, automatable, tactical, devoid of enduring value, and that scales linearly as a service grows.
+This blog post describes our journey to automate our nodes rotation process when we have a new AMI release, and the open source tools we built on this.
+
+#### Problems
+
+Apart from toil elimination we had specific problems that we needed to solve by building our tooling around.
+Our first problem is the limited way in which kops rolls out node changes. It does it sequentially, one by one and in a specific window of time. We needed a much more flexible way of rotation, to avoid stretching our workloads or hitting any limits on our infrastructure. Each environment might need different handling to be more efficient and as reliable as possible during this process.
```

Comment:
```suggestion
Our first problem is the limited way in which Kubernetes Operations (kops) rolls out node changes. It does it sequentially, one by one and in a specific window of time. We needed a much more flexible way of rotation, to avoid stretching our workloads or hitting any limits on our infrastructure. Each environment might need different handling to be more efficient and as reliable as possible during this process.
```